libres/lib/enkf/enkf_main.cpp

```diff
@@ -1134,13 +1318,13 @@ bool enkf_main_smoother_update(enkf_main_type *enkf_main,
                                enkf_fs_type *target_fs) {
     time_map_type *time_map = enkf_fs_get_time_map(source_fs);
 
-    int step2 = time_map_get_last_step(time_map);
-    if (step2 < 0)
-        step2 = model_config_get_last_history_restart(
+    int last_step = time_map_get_last_step(time_map);
+    if (last_step < 0)
+        last_step = model_config_get_last_history_restart(
             enkf_main_get_model_config(enkf_main));
 
     int_vector_type *step_list = int_vector_alloc(0, 0);
-    for (int i = 0; i <= step2; i++)
+    for (int i = 0; i <= last_step; i++)
         int_vector_append(step_list, i);
 
     bool update_done = enkf_main_UPDATE(enkf_main, step_list, source_fs,
```

Comment:
This seems to be the only call to `enkf_main_UPDATE` and the parameter `run_mode` is set to `SMOOTHER_RUN`.
Should we remove it as a parameter, or do we need the flexibility for something?

Reply:
more cleanup should be done, but maybe not in this PR

Reply:
Agree, I'll create an issue.